cores/esp8266/Esp.cpp

```diff
@@ -444,21 +444,23 @@ bool EspClass::checkFlashConfig(bool needsEquals) {
     return false;
 }
 
+extern "C" uint32_t __crc_len;
+extern "C" uint32_t __crc_val;
 bool EspClass::checkFlashCRC() {
     // The CRC and total length are placed in extra space at the end of the 4K chunk
     // of flash occupied by the bootloader.  If the bootloader grows to >4K-8 bytes,
     // we'll need to adjust this.
-    uint32_t flashsize = *((uint32_t*)(0x40200000 + 4088)); // Start of PROGMEM plus 4K-8
-    uint32_t flashcrc = *((uint32_t*)(0x40200000 + 4092)); // Start of PROGMEM plus 4K-4
+    uint32_t flashsize = __crc_len;
+    uint32_t flashcrc = __crc_val;
     uint32_t z[2];
     z[0] = z[1] = 0;
 
     // Start the checksum
-    uint32_t crc = crc32((const void*)0x40200000, 4096-8, 0xffffffff);
+    uint32_t crc = crc32((const void*)0x40200000, 4096 + 16, 0xffffffff);
     // Pretend the 2 words of crc/len are zero to be idempotent
     crc = crc32(z, 8, crc);
     // Finish the CRC calculation over the rest of flash
-    crc = crc32((const void*)0x40201000, flashsize-4096, crc);
+    crc = crc32((const void*)0x40201018, flashsize - 4096 - 16 - 8, crc);
```

Comment:
btw, we already know the 'location' via `&__crc_len` aka `0x40201010`. may these should use locations of vars + their size?
(or `&_irom0_text_start` with is also the start of the section. and, I think linker already creates `__irom0_text_start`? or something like that for each section automatically)

And I wonder if python script can somehow extract those off of the .elf instead of `4096 + 16`

Reply:
Good idea.  In the core I think that can be done. simply enough.  I'll throw an update.

In the Python script, accessing the symbol table would require scraping `objdump` output or something, which I think would be less clear and more prone to errors than what's there now.